Sources/Transaction.swift

```diff
@@ -10,13 +10,40 @@ import BigInt
 public struct Transaction {
     public var accountNonce: UInt64
     public var price: BigInt
-    public var gasLimit: BigInt
-    public var recipient: Address?
+    public var gasLimit: UInt64
+    public var recipient: Address
     public var amount: BigInt
     public var payload: Data?
 
     // Signature values
-    public var v: BigInt
-    public var r: BigInt
-    public var s: BigInt
+    public var v = BigInt()
+    public var r = BigInt()
+    public var s = BigInt()
+
+    /// Creates a `Transaction` with default values and the given recipient address.
+    public init(recipient: Address) {
+        accountNonce = 0
+        price = BigInt()
+        gasLimit = 0
+        self.recipient = recipient
+        amount = BigInt()
+    }
+
+    /// Signs this transaction by filling in the `v`, `r`, and `s` values.
+    ///
+    /// - Parameters:
+    ///   - chainID: chain identifier, defaults to `1`
+    ///   - hashSigner: function to use for signing the hash
+    public mutating func sign(chainID: Int = 1, hashSigner: (Data) throws -> Data) rethrows {
```

Comment:
Honestly, I would ditch `HomesteadSigner` at all. everyone use `EIP155` for this stuff.

Reply:
Again, this is based on geth. I don't think it's harmful to have.